src/SimplePie.php

```diff
@@ -828,6 +852,14 @@ public function set_curl_options(array $curl_options = [])
      */
     public function force_fsockopen(bool $enable = false)
     {
+        if ($this->http_client_injected) {
+            throw new SimplePieException(sprintf(
+                'Using "%s()" has no effect, because you already provided a HTTP client with "%s::set_http_client()". Configure fsockopen in your HTTP client instead.',
+                __METHOD__,
+                self::class
+            ));
+        }
+
```

Comment:
We should unset a possible created $this->http_client on calling `force_fsockopen()`.
```suggestion

        // Reset a possible existing FileClient,
        // so a new client with the changed value will be created
        $this->http_client = null;
```